airbyte-integrations/connectors/source-prestashop/source_prestashop/components.py

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import datetime
+from dataclasses import InitVar, dataclass, field
+from typing import Any, ClassVar, Iterable, List, Mapping, Optional, Union
+
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from airbyte_cdk.sources.declarative.requesters.paginators.paginator import Paginator
+from airbyte_cdk.sources.declarative.requesters.paginators.strategies.pagination_strategy import PaginationStrategy
+from airbyte_cdk.sources.declarative.requesters.request_option import RequestOption
+from airbyte_cdk.sources.declarative.stream_slicers.stream_slicer import StreamSlicer
+from airbyte_cdk.sources.declarative.types import Config, Record, StreamSlice, StreamState
+from dataclasses_jsonschema import JsonSchemaMixin
+
+
+@dataclass
+class LimitOffsetPaginator(Paginator, JsonSchemaMixin):
```

Comment:
Could you please add some docstrings explainig how things work and unit tests for the custom components?

Reply:
+1. It would be useful to document why the custom components are needed. Ideally we can enhance the existing ones so they can support this integration instead

Reply:
LimitOffsetPaginator - very similar to DefaultPaginator
the only difference:

DefaultPaginator has 2 params: `page_size_option` and `page_token_option`

but 

LimitOffsetPaginator has only 1 param: `limit_offset_option`

Reply:
Ah that's a fun one! It should be a fairly small lift to enhance the framework to support this new feature
1. Make the DefaultPaginator's page_token_option optional
2. Add the request parameter to the `request_options_provider` (will probably look like `"'{{next_page_token.next_page_token}},{{options.page_size}}'"`)

Reply:
https://github.com/airbytehq/airbyte/pull/19368

Reply:
@girarda I have closed #19368 and re-implemented using `next_page_token['next_page_token']`